zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -157,8 +160,8 @@ public void run() {
 
     private ChangeRecord getRecordForPath(String path) throws KeeperException.NoNodeException {
         ChangeRecord lastChange = null;
-        synchronized (zks.outstandingChanges) {
-            lastChange = zks.outstandingChangesForPath.get(path);
+        synchronized (zks.getOutstandingChanges()) {
```

Comment:
Can you call the getter outside the synchronized block and then. Use the reference to the returned object?
It will be clearer that we are accessing the same object

Reply:
I wasn't feeling comfortable doing this either. Even calling the getter outside is not satisfying because this object is being used in many places. How about we add the field in PrepRequestProcessor and make it final and initialize it in the constructor?

Reply:
Okay